lib/client.js

```diff
@@ -467,7 +467,14 @@ function Request(method, url) {
       self.emit('response', res);
     }
 
-    self.callback(err, res);
+    if (res && res.status >= 200 && res.status < 300) {
+      return self.callback(err, res);
+    }
+
+    var new_err = new Error('Unsucessful HTTP response');
```

Comment:
Why? You know if it is a network error or not based on if there is a `response` object in the callback.

We could add a flag to the `error` object to let you know more but my thinking was that just checking if you have a response (after error) would let you know more about the situation.